ui/raidboss/data/06-ew/raid/p4s.ts

```diff
@@ -72,16 +83,50 @@ const roleOutputStrings = {
   unknown: Outputs.unknown,
 };
 
+const tetherOutputStrings = {
+  purpleTether: {
+    en: 'Purple Tether',
+  },
+  orangeTether: {
+    en: 'Orange Tether',
+  },
+  greenTether: {
+    en: 'Green Tether',
+  },
+  blueTether: {
+    en: 'Blue Tether',
+  },
+};
+
+const curtainCallOutputStrings = {
+  tetherGroup: {
+    en: 'Tether Group ${num}',
+  },
+  groupTethers: {
+    en: 'Group ${num} Tethers',
+  },
+};
+
 // Due to changes introduced in patch 5.2, overhead markers now have a random offset
 // added to their ID. This offset currently appears to be set per instance, so
 // we can determine what it is from the first overhead marker we see.
-// The first 1B marker in the encounter is an Elegant Evisceration? (00DA?).
+// The first 1B marker in the encounter is an Elegant Evisceration (00DA).
+// The first 1B marker in the phase 2 encounter could be one of three:
+// Purple, Green, Orange color tethers (012D, 012E, 012F)
 const firstHeadmarker = parseInt('00DA', 16);
+const purpleMarker = parseInt('012D', 16);
+
 const getHeadmarkerId = (data: Data, matches: NetMatches['HeadMarker']) => {
   // If we naively just check !data.decOffset and leave it, it breaks if the first marker is 00DA.
   // (This makes the offset 0, and !0 is true.)
-  if (typeof data.decOffset === 'undefined')
-    data.decOffset = parseInt(matches.id, 16) - firstHeadmarker;
+  if (typeof data.decOffset === 'undefined') {
+    if (typeof data.act === 'undefined') {
+      data.decOffset = parseInt(matches.id, 16) - firstHeadmarker;
+    } else {
+      data.colorHeadmarkerIds ??= [];
+      data.decOffset = (data.colorHeadmarkerIds[0] ?? 0) - purpleMarker;
```

Comment:
I'm not sure what this code is doing.  Doesn't `P4S Color Headmarkers` set this too?  I think only one place should do it.

Reply:
It isn't run, I'll remove that. The phase 2 setting of decOffset is occuring outside of the function here.